syscalls.go

```diff
@@ -301,6 +312,25 @@ func bpfMapGetNextKey(m *internal.FD, key, nextKeyOut internal.Pointer) error {
 	return wrapMapError(err)
 }
 
+func objGetNextID(cmd int, start uint32) (uint32, error) {
+	attr := bpfObjGetNextIDAttr{
+		startID: start,
+	}
+	_, err := internal.BPF(cmd, unsafe.Pointer(&attr), unsafe.Sizeof(attr))
+	return attr.nextID, wrapObjError(err)
+}
+
+func wrapObjError(err error) error {
+	if err == nil {
+		return nil
+	}
+	if xerrors.Is(err, unix.ENOENT) {
+		return ErrNotExist
```

Comment:
This needs to be wrapped.

```suggestion
		return xerrors.Errorf("%w", ErrNotExist)
```

Reply:
Following `wrapMapError()`, where we also directly return `ErrKeyNotExist`, I did not wrap this error. Also, just like `ErrKeyNotExist` `ErrNotExist` becomes part of the API. Do you want to avoid this? If we wrap this error here, I also need to adapt the comments for `MapGetNextID` and `ProgramGetNextID`.

Reply:
Yeah, I want to avoid unwrapped errors, to be able to add or remove detail later on. All (indirect) callers of wrapMapError end up returning a wrapped error, so there never should be a bare ErrKeyNotExist. The same should apply for ErrNotExist.

Like I mentioned, the wrap* functions are really confusing and probably need to go. But that can happen later.